### PR TITLE
Fixes #10605 - handle logout accessed by GET

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

File: test/functional/users_controller_test.rb [hunk 1]
```diff
@@ -210,7 +210,7 @@ class UsersControllerTest < ActionController::TestCase
     @controller.stubs(:available_sso).returns(@sso)
     @controller.stubs(:get_sso_method).returns(@sso)
     get :extlogin, {}, {}
-    get :logout, {}, {}
+    post :logout, {}, {}
```
Comment (ares) [inline test/functional/users_controller_test.rb:213]: fixes test that worked accidentally

Comment (iNecas) [inline test/functional/users_controller_test.rb:213]: What was the accident? How do we prevent from happening again?

Comment (ares) [inline test/functional/users_controller_test.rb:213]: > What was the accident?

GET for :logout didn't exist until now. Functional tests don't go through routes so even PUT :logout would work. But now when GET got a different logic it stopped redirecting the user.

> How do we prevent from happening again?

not sure, how about stop writing functional tests? :-)

File: app/views/users/logout.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<h2><%= _('Are you sure you want to logout?') %></h2>
```
Comment (dLobatog) [inline app/views/users/logout.html.erb]: "log out"

File: app/views/users/logout.html.erb [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+<%= button_to _('yes'), logout_users_url, :class => 'btn btn-primary' %>
```
Comment (dLobatog) [inline app/views/users/logout.html.erb]: 'Yes' (capital Y) - I would also add a 'No' option, which goes back to the previous page?

Comment (ares) [inline app/views/users/logout.html.erb]: The reason this page is being added is that if you access the URL directly (e.g. bookmarked log out link from menu) you'd get 404. There's no way to get to this page from Foreman navigation, so there's no "previous" page. I could add "No" link that would always point to let's say dashboard or hosts index...

**Comment by ares** [comment]
I think all comments were addressed.

**Comment by dLobatog** [comment]
Merged as 3f579c3e14459ac93446265473ef4e1fdb13256e, thanks @ares!

**Pull request closed without merging**